content/TypeHinting/TypeHinting.q.md

```diff
@@ -0,0 +1,1346 @@
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+declare(strict_types = 1);
+
+class Person 
+{
+  protected $name;
+
+  public function __construct(string $name)
+  {
+    $this->name = $name;
+  }
+
+  public function setName(string $name)
+  {
+    $this->name = $name;
+  }
+
+  public function getName() : string
+  {
+    return $this->name;
+  }
+}
+
+class Car
+{
+  protected $driver;
+
+  public function __construct(Person $driver)
+  {
+    $this->driver = $driver;
+  }
+}
+
+$personObject = new Person("John");
+$myCar = new Car("$personObject");
+print_r($myCar);
+?>
+```
+/// type=MS, answer=[2,5]
+
+Execute the program. What are the error messages?
+
+ - Uncaught Error: Call to undefined function `Person()` on line number 34
+
+ - Object of class `Person` could not be converted to string on line number 35
+
+ - Uncaught TypeError: Argument `1` passed to `Person::setName()` must be of the type string, none given, called on line 20 and defined on line number 8
+
+ - Uncaught TypeError: Argument `1` passed to `Person::__construct()` must be of the type string, none given, called on line 34 and defined on line number 8
+
+ - Uncaught TypeError: Argument `1` passed to `Car::__construct()` must be an instance of Person, string given, called on line 35 and defined on line number 28
+
+
+/// type=MS, answer=[2,3,5]
+
+Which statements correctly describe the error?
+
+ - There is no semicolon `;` at the end of the statement on line 34.
+
+ - On line 35, the argument `$personObject` is enclosed in double quotes `""`.
+
+ - On line 35, the statement `$myCar = new Car("$personObject");` is invalid.
+
+ - There is no `new` keyword specified before `Car("$personObject")` on line 35.
+
+ - The `__construct()` method of the `Car` class requires an argument of type `Object`.
+
+:::
+
+
+/// type=CR, answer=[tests/TypeHinting/RemoveUnwantedDoubleQuotesTest.php]
+
+Correct the code so that it outputs the string `Car Object ( [driver:protected] => Person Object ( [name:protected] => John ) )`.
+
+```php
+<?php
+declare(strict_types = 1);
+
+class Person 
+{
+  protected $name;
+
+  public function __construct(string $name)
+  {
+    $this->name = $name;
+  }
+
+  public function setName(string $name)
+  {
+    $this->name = $name;
+  }
+
+  public function getName() : string
+  {
+    return $this->name;
+  }
+}
+
+class Car
+{
+  protected $driver;
+
+  public function __construct(Person $driver)
+  {
+    $this->driver = $driver;
+  }
+}
+
+$personObject = new Person("John");
+$myCar = new Car("$personObject");
+print_r($myCar);
+?>
+```
+
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/TypeHinting/CreateMethodsWithTypeDeclarationsTest.php]
+
+Write a program that adds type declarations to methods of a certain class. First, add a statement `declare(strict_types = 1);` after the `<?php` tag. Then, use a `class` keyword to declare a class named `Animal`. Within the curly braces `{}`, add the following statements:
```

Comment:
You mentioned before that we still have issue rendering content in lists. Can we list instructions now?

Reply:
Yes, but this is in connection with OOP lesson wherein we have to separate each instruction into lists since we are already dealing with classes and it is expected to have a longer instructions/statements. This is also part of T's request.

Reply:
Noted, thank you.